index.js

```diff
@@ -166,7 +204,13 @@ export class UpdateAPK {
     const result = data.results[0];
     const version = result.version;
     const trackViewUrl = result.trackViewUrl;
-    if (version !== RNUpdateAPK.versionName) {
+
+    let updateAvailable = version !== RNUpdateAPK.versionName
+    if(this.options.compareVersionNameByParts){
+      updateAvailable = this.checkVersion(RNUpdateAPK.versionName,version) === -1
+    }
```

Comment:
nitpick: fixed spacing
logic comment: shouldn't this be `> 0` to be more correct?

```suggestion
    if (this.options.compareVersionNameByParts) {
      updateAvailable = this.checkVersion(RNUpdateAPK.versionName,version) < 0
    }
```